File: Data Structure/Tree/Binary search tree/Traverse/Breadth first search/iterative.cpp
int Height(struct Node* head){
        if(head == NULL){
                return 0;
        }
        return 1 + max(Height(head->right), Height(head->left));
}


void printOrder(Node* head, int level) {
        if(head == NULL) return;
        if(level == 1) {
                cout << head->val << endl;
        };
        if(level > 1) {
                printOrder(head->left, level - 1);
                printOrder(head->right, level - 1);
        }
}


void printLevelOrder(Node* head) {
        int h = Height(head);
        for(int i = 1; i <= h; i++) {
                printOrder(head, i);
        }
}

File: Data Structure/Tree/Binary search tree/Traverse/Breadth first search/iterative.js
/*
class Node{
        constructor(value) {
                this.val = value;
                this.left = null;
                this.right = null;
        }
};
*/


var levelOrder = function (root) {
        if (root == null) return [];
        const queue = [];
        let count;
        let temp;
        queue.push(root);
        while(queue.length != 0) {
                count = queue.length;
                for(let i = 0; i < count; i++) {
                        temp = queue.shift();
                        console.log(temp.val);
                        if(temp.left != null) queue.push(temp.left);
                        if(temp.right != null) queue.push(temp.right);
                }
        }
};      
